### PR TITLE
Add search-astro-docs extension

### PR DESCRIPTION
**Pull request by ElianCodes**
## Description

Searches [The Astro documentation](https://docs.astro.build) from Raycast

## Screencast

https://github.com/raycast/extensions/assets/15145918/e8970dd4-05dd-48d8-8dbc-12858033000e

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Hi 👋 

Thanks for your contribution 🔥 

It would be nice if it actually searched the documentation, a bit like [this extension](https://www.raycast.com/danyelkeddah/spatie-documentation) and many of the other documentation extensions does it.

If that's not possible, then I would make it a `no-view` command with an argument instead 🙂

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by ElianCodes** [comment]
Hey @pernielsentikaer! Thanks for your ideas!

I've updated the extension to work with our multilingual documentation as well. Also filled the list with all of our bigger documentation pages. Later on, I might also include some more.
There is a fallback for pages that are not listed in the list, but still will work through https://a.stro.cc.

I've also included future-proofing, like adding icons and keywords to the list items themselves. Which we might come back to edit later.

https://github.com/raycast/extensions/assets/15145918/e8beca44-66b7-49c9-8dc7-9551e80274f4

File: extensions/search-astro-docs/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        }
+      />
+    </>
+  );
+}
+
+function OpenSearchInBrowserAction(props: { search: string }): ReactElement {
+  const url = new URL(`https://a.stro.cc/${props.search}`);
+  return <Action.OpenInBrowser title="Search on Astro Documentation" icon="astro-search-icon.png" url={url.href} />;
+}
```
Comment (pernielsentikaer) [inline extensions/search-astro-docs/src/index.tsx:102]:
```suggestion
import { Action, ActionPanel, Icon, List, openExtensionPreferences } from "@raycast/api";
import { useState } from "react";
import { URL } from "url";
import { documentationListV3 as docsList } from "./data/docs";

export default function UserSearchRoot() {
  const [search, setSearch] = useState<string>();
  return (
    <List searchBarPlaceholder="Search the Astro Documentation" onSearchTextChange={setSearch} filtering={true}>
      <List.EmptyView
        title="No Results"
        description={`Search '${search}' in the Astro Documentation`}
        actions={
          <ActionPanel>
            <OpenSearchInBrowserAction search={search ?? ""} />
          </ActionPanel>
        }
      />
      {docsList.map((docsItem) => (
        <List.Item
          keywords={docsItem.keywords}
          key={docsItem.title}
          title={`${docsItem.title}`}
          icon={docsItem.icon ?? "astro-search-icon.png"}
          actions={
            <ActionPanel>
              <Action.OpenInBrowser
                title="Open the Astro Documentation"
                url={docsItem.url ? docsItem.url : `http://a.stro.cc/${docsItem.title}`}
              />
              <Action
                title="Open Extension Preferences"
                onAction={openExtensionPreferences}
                icon={Icon.Gear}
                shortcut={{ modifiers: ["cmd", "shift"], key: "," }}
              />
            </ActionPanel>
          }
        />
      ))}
    </List>
  );
}

function OpenSearchInBrowserAction(props: { search: string }) {
  const url = new URL(`https://a.stro.cc/${props.search}`);
  return <Action.OpenInBrowser title="Search on Astro Documentation" icon="astro-search-icon.png" url={url.href} />;
}
```
What about doing something like this instead?

Comment (ElianCodes) [reply extensions/search-astro-docs/src/index.tsx]: Ah! This is a great suggestion! Thanks! didn't know that `Icon` and `filtering` existed. Thank you so much. I've applied your suggestions in a new commit!

File: extensions/search-astro-docs/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          <ActionPanel>
+            <Action
+              title="Open Extension Preferences"
+              onAction={openExtensionPreferences}
+              icon={Icon.Gear}
+              shortcut={{ modifiers: ["cmd", "shift"], key: "," }}
+            />
+          </ActionPanel>
+        }
+      />
```
Comment (pernielsentikaer) [inline extensions/search-astro-docs/src/index.tsx:51]: There is no other place in Raycast that adds it as a List.Item, I would add it as an action on the looped items instead 🙂

Comment (ElianCodes) [reply extensions/search-astro-docs/src/index.tsx:51]: When I add it in the loop, it doesn't show up for me when I do a search for it

Comment (pernielsentikaer) [reply extensions/search-astro-docs/src/index.tsx:51]: It will only be available in the action panel, do you like having it in the list instead?

**Comment by ElianCodes** [comment]
@pernielsentikaer, I've implemented all the required behaviour back into [the original repo](https://github.com/ElianCodes/search-astro-docs). Maybe it's a little easier to work from there.

The required behaviour is basically:
- the loop with all openable documentation links (with language support)
- an item that opens the plain documentation (https://docs.astro.build)
- any search that doesn't match an item has to go through `https:a.stro.cc/<searchitem>`
- open preferences

File: extensions/search-astro-docs/src/index.tsx (hunk 2 above)
Comment (ElianCodes) [reply extensions/search-astro-docs/src/index.tsx:60]: Yes! I would like it to be in the list instead. I think it's more accessible / easier that way

**Comment by pernielsentikaer** [comment]
Since no other view in Raycast has "Open Preferences" in the list, it will really confuse the end user. They are used to finding it in the action panel. You can also make the language a required field, so the user can set it initially when downloading it from the store.

**Comment by pernielsentikaer** [comment]
What about
```
import { Action, ActionPanel, Icon, List, openExtensionPreferences } from "@raycast/api";
import { useState } from "react";
import { URL } from "url";
import { documentationListV3 as docsList } from "./data/docs";

export default function UserSearchRoot() {
  const [search, setSearch] = useState<string>();
  return (
    <List searchBarPlaceholder="Search the Astro Documentation" onSearchTextChange={setSearch} filtering throttle>
      {docsList.map((docsItem) => (
        <List.Item
          keywords={docsItem.keywords}
          key={docsItem.title}
          title={`${docsItem.title}`}
          icon={docsItem.icon ?? "astro-search-icon.png"}
          actions={
            <ActionPanel>
              <Action.OpenInBrowser
                title="Open the Astro Documentation"
                url={docsItem.url ? docsItem.url : `http://a.stro.cc/${docsItem.title}`}
              />
              <Action onAction={openExtensionPreferences} title="Open Extension Preferences" icon={Icon.Gear} />
            </ActionPanel>
          }
        />
      ))}
      <List.Item
        icon={"astro-search-icon.png"}
        title={`Search '${search}' in the Astro Documentation`}
        actions={
          <ActionPanel>
            <OpenSearchInBrowserAction search={search ?? ""} />
          </ActionPanel>
        }
      />
      <List.Item
        key={"open Astro documentation"}
        title={"Open the Astro Documentation"}
        icon={"astro-search-icon.png"}
        actions={
          <ActionPanel>
            <Action.OpenInBrowser title="Open the Astro Documentation" url={"http://docs.astro.build/"} />
          </ActionPanel>
        }
      />
    </List>
  );
}

function OpenSearchInBrowserAction(props: { search: string }) {
  const url = new URL(`https://a.stro.cc/${props.search}`);
  return <Action.OpenInBrowser title="Search on Astro Documentation" icon="astro-search-icon.png" url={url.href} />;
}
```
I added throttle since it made a lot of empty state flicker while searching, and I moved the extension preference to action panel, (could be added to all items in the list) - Could that work for you?

**Comment by ElianCodes** [comment]
Oh yes @pernielsentikaer! That makes sense! I've updated the extension / repo!
Thanks so much for the review and suggestions ❤️

**Comment by pernielsentikaer** [comment]
Are you okay with the last contribution I made 🙂 Now we always have `Open the Astro Documentation` and `Search '${search}' in the Astro Documentation` last in the list

**Comment by ElianCodes** [comment]
I think that's fine! After we have some members using it, I'll gather some feedback and update to a new version accordingly

**Comment by pernielsentikaer** [comment]
Awesome, let's ship it 🚀 

Thanks for your cooperation, it was a pleasure working with you 🫶

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**